java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java

```diff
@@ -1551,6 +1542,44 @@ private void readStreetsAndBuildingsForCity(PreparedStatement streetBuildingsSta
 		set.close();
 	}
 
+	private Street addStreetToUniqueNamesMap(Map<String, List<Street>> uniqueNames, String streetName, Map<String, String> names, City city) {
+		Street street = new Street(city);
+		if (!uniqueNames.containsKey(streetName)) {
+			String localeStreetName = uniqueNames.keySet()
```

Comment:
code overcomplicated: 2 for + inner streams - basically 3 for loops inside and no clear names what's going on